File: backend/src/main/java/com/financemate/account/model/Account.java
package com.financemate.account.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "accounts")
public class Account {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;
    private String name;
    private String description;
    private String userId;
    private String currencyCode;
    private double balance;
    private String color;
    private boolean includeInStats;
    private boolean archived;

}

File: frontend/src/types/account.ts
interface AccountBase {
    name: string
    description?: string
    balance: number
    color: string
}

export interface Account extends AccountBase {
    id: string
    includeInStats: boolean
    archived?: boolean
}

export interface AccountDto extends AccountBase {}

export interface TransferDto {
    fromAccountId?: string
    toAccountId?: string
    amount: number
}